### DISCUSSION TITLE
Is it possible to have different parameter examples for each endpoint?

### DISCUSSION THREAD

**Discussion by jeremyschiemann** [fastapi/fastapi#8760] [Questions] [answered]
### Description

Hi, 
the title pretty much says it all, I'm trying to have different examples for different endpoints in the swagger ui but with the same base parameters.
In my example I have a common 'weight' parameter and would like to have the example '3' for cats and the '10'  from the Base for everything else. 
Might not be the best example but I think it helps understanding what I'm trying to do.

With the example code, both endpoints will show '10' as example.

Thanks in advance :)

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from fastapi import FastAPI, Depends
import uvicorn
from fastapi import Query
from pydantic.dataclasses import dataclass


@dataclass
class TestModelBase:
    weight: int = Query(..., example=10, description="Weight of an animal.")


class TestModelDogs(TestModelBase):
    pass


class TestModelCats(TestModelBase):
    weight: int = Query(..., example=3)


app = FastAPI()


@app.get("/Dogs")
def test1(a: TestModelDogs = Depends()) -> str:
    return "Hello World Dogs"


@app.get("/Cats")
def test2(a: TestModelCats = Depends()) -> str:
    return "Hello Worlds Cats"


if __name__ == "__main__":
    uvicorn.run(app='app.__main__:app', reload=True, access_log=False)

```

### Environment

* OS: Windows 10 with WSL2 Ubuntu
* FastAPI Version: 0.67.0

**Comment by YuriiMotov** [comment, marked as answer]
The following works fine in current version (0.115.13).

```python
from typing import Annotated
from fastapi import FastAPI
from pydantic import BaseModel
from fastapi import Query


class TestModelBase(BaseModel):
    weight: int = Query(..., example=10, description="Weight of an animal.")


class TestModelDogs(TestModelBase):
    pass


class TestModelCats(TestModelBase):
    weight: int = Query(..., example=3)


app = FastAPI()


@app.get("/Dogs")
def test1(a: Annotated[TestModelDogs, Query()]) -> str:
    return "Hello World Dogs"


@app.get("/Cats")
def test2(a: Annotated[TestModelCats, Query()]) -> str:
    return "Hello Worlds Cats"

```

See Docs: https://fastapi.tiangolo.com/tutorial/query-param-models/
